targets/TARGET_STM/gpio_object.h

```diff
@@ -61,7 +61,7 @@ static inline void gpio_write(gpio_t *obj, int value)
         *obj->reg_set = obj->mask;
     } else {
 #ifdef GPIO_IP_WITHOUT_BRR
-        *obj->reg_clr = obj->mask << 16;
+        *obj->reg_set = obj->mask << 16;
 #else
```

Comment:
Found my problem,   I did not do a full clean and rebuild so gpio_api did not pickup the change.